infrastructure/persistence/src/main/kotlin/com/studentcenter/weave/infrastructure/persistence/meetingTeam/repository/MeetingTeamJpaRepository.kt

```diff
@@ -30,4 +30,37 @@ interface MeetingTeamJpaRepository : JpaRepository<MeetingTeamJpaEntity, UUID> {
     ): List<MeetingTeamJpaEntity>
 
 
+    // TODO : JOOQ 설정 이후 변경 필요
+    @Query(
+        value =
+        """
+            SELECT mt.id, mt.team_introduce, mt.member_count, mt.location, mt.status, mt.gender
+            FROM (
+                SELECT *
+                FROM meeting_team
+                WHERE (:memberCount IS NULL OR member_count = :memberCount)
+                  AND (:next IS NULL OR id > :next)
+                  AND (:status IS NULL OR status = :status)
+                  AND (:gender IS NULL OR gender = :gender)
+                  AND (:preferredLocations IS NULL OR location IN (:preferredLocations))
+            ) mt
+            INNER JOIN weave.meeting_team_member_summary mtms ON mt.id = mtms.meeting_team_id
+            WHERE (mtms.youngest_member_birth_year <= :youngestMemberBirthYear)
+              AND (mtms.oldest_member_birth_year >= :oldestMemberBirthYear)
+            ORDER BY mt.id DESC
+            LIMIT :limit
```

Comment:
제가 생각한 느낌은 이런 느낌이었어요! summary에 member count 있다고 착각했네요!
```suggestion
            SELECT mt.id, mt.team_introduce, mt.member_count, mt.location, mt.status, mt.gender
            FROM meeting_team mt
            INNER JOIN (
              SELECT meeting_team_id
              FROM weave.meeting_team_member_summary
              WHERE (mtms.youngest_member_birth_year <= :youngestMemberBirthYear)
              AND (mtms.oldest_member_birth_year >= :oldestMemberBirthYear)
            ) mtms ON mt.id = mtms.meeting_team_id
            WHERE (:memberCount IS NULL OR member_count = :memberCount)
            AND (:next IS NULL OR id > :next)
            AND (:gender IS NULL OR gender = :gender)
            AND (:preferredLocations IS NULL OR location IN (:preferredLocations))
            ORDER BY mt.id DESC
            LIMIT :limit
```

Reply:
이렇게 하소 최소/최대/미팅 팀 id 로 covering index!

Reply:
제가 작성한 쿼리는 meeting_team의 컬럼을 기준으로 한번 걸러내는거고, 보여주신 쿼리는 meeting_team_summary의 컬럼을 기준으로 한번 걸러내고 시작하는것으로 보이는데, 일반적인 사용사례를 생각했을때, meeting_team에 있는 gender컬럼은 최소한 절반이상 거르고 시작해서 meeting_team 컬럼 기준으로 걸러내는게 좋지 않을까요?

meeting_team 컬럼 조건으로 우선 걸러내는게 훨씬 많이 걸러내질것 같은데 어떻게 생각하시나요? 
```sql
CREATE INDEX idx_meeting_team_member_summary_filter ON meeting_team_member_summary(meeting_team_id, youngest_member_birth_year, oldest_member_birth_year);
```
기존방식에서도 말씀해주신대로 인덱스 걸어도 인덱스를 타서 문제없을것 같아요

```
1,SIMPLE,meeting_team,,index,PRIMARY,PRIMARY,16,,1,100,Using where; Backward index scan
1,SIMPLE,mtms,,eq_ref,"meeting_team_id,idx_meeting_team_member_summary_filter",meeting_team_id,16,weave.meeting_team.id,1,100,Using where
```

(추가적으로 meeting_team에는 커버링 인덱스를 걸수가 없네요 키 크기 제한에 걸려서)

Reply:
커버링 인덱스가 안된다면 기존 방식이 좋아보입니다!
커버링 인덱스를 하려고 했던 이유는 summary 쪽 데이터를 일부만 사용하기 때문에 인덱스로 빠르게 처리가능하다면 그게 좋다고 생각했어요!